src/api/Tron.js

```diff
@@ -0,0 +1,529 @@
+// @flow
+import { BigNumber } from "bignumber.js";
+import type {
+  Transaction,
+  TrongridTxInfo,
+  SendTransactionData,
+  SendTransactionDataSuccess,
+  SmartContractTransactionData,
+  FreezeTransactionData,
+  UnfreezeTransactionData,
+  NetworkInfo,
+  BandwidthInfo,
+  SuperRepresentative,
+  SuperRepresentativeData,
+  TronResources
+} from "../families/tron/types";
+import type { Account, SubAccount, Operation } from "../types";
+import {
+  decode58Check,
+  encode58Check,
+  abiEncodeTrc20Transfer,
+  formatTrongridTxResponse,
+  hexToAscii
+} from "../families/tron/utils";
+import { log } from "@ledgerhq/logs";
+import network from "../network";
+import { retry } from "../promise";
+import { makeLRUCache } from "../cache";
+import { getEnv } from "../env";
+import get from "lodash/get";
+import drop from "lodash/drop";
+import sumBy from "lodash/sumBy";
+import take from "lodash/take";
+
+const baseApiUrl: string = getEnv("API_TRONGRID_PROXY");
+
+async function post(url: string, body: Object) {
+  const { data } = await network({
+    method: "POST",
+    url,
+    data: body
+  });
+
+  // Ugly but trongrid send a 200 status event if there are errors
+  if (data.Error) throw new Error(data.Error);
+
+  return data;
+}
+
+async function fetch(url: string) {
+  const { data } = await network({
+    method: "GET",
+    url
+  });
+
+  // Ugly but trongrid send a 200 status event if there are errors
+  if (data.Error) throw new Error(data.Error);
+
+  return data;
+}
+
+export const freezeTronTransaction = async (
+  a: Account,
+  t: Transaction
+): Promise<SendTransactionDataSuccess> => {
+  const txData: FreezeTransactionData = {
+    frozen_balance: t.amount.toNumber(),
+    frozen_duration: t.duration || 3,
+    resource: t.resource,
+    owner_address: decode58Check(a.freshAddress),
+    receiver_address: t.recipient ? decode58Check(t.recipient) : undefined
+  };
+
+  const url = `${baseApiUrl}/wallet/freezebalance`;
+
+  const result = await post(url, txData);
+
+  return result;
+};
+
+export const unfreezeTronTransaction = async (
+  a: Account,
+  t: Transaction
+): Promise<SendTransactionDataSuccess> => {
+  const txData: UnfreezeTransactionData = {
+    resource: t.resource,
+    owner_address: decode58Check(a.freshAddress),
+    receiver_address: t.recipient ? decode58Check(t.recipient) : undefined
+  };
+
+  const url = `${baseApiUrl}/wallet/unfreezebalance`;
+  const result = await post(url, txData);
+
+  return result;
+};
+
+// Send trx or trc10/trc20 tokens
+export const createTronTransaction = async (
+  a: Account,
+  t: Transaction,
+  subAccount: ?SubAccount
+): Promise<SendTransactionDataSuccess> => {
+  const [tokenType, tokenId] =
+    subAccount && subAccount.type === "TokenAccount"
+      ? drop(subAccount.token.id.split("/"), 1)
+      : [undefined, undefined];
```

Comment:
that part is a bit obscure. I recommend to write a helper function that split out the id components. (similarly to what you can see in accountId.js)